src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -34,6 +35,12 @@ namespace Microsoft.CodeAnalysis.CodeActions
     /// </summary>
     public abstract class CodeAction
     {
+        /// <summary>
+        /// Special tag that indicates that it's this is a privileged code action that is allowed to use the <see
+        /// cref="CodeActionPriority.High"/> priority class.
+        /// </summary>
+        internal static readonly string CanBeHighPriorityTag = Guid.NewGuid().ToString();
```

Comment:
this approach ensures one can't just learn the special tag and include it themselves.  they would ahve to use reflection to find and use this.  and, at that point, all bets are off.  if they're willing to do that, we can't stop them in any event from becoming high pri.

Reply:
Your comment makes me feel that we should in fact do this in the CodeStyle layer - have a special helper that uses reflection to add this custom tag to all our high priority providers. This way we stay with your current approach in this PR, and at the same time users don't see any behavioral differences when adding or removing a NuGet package reference to the CodeStyle package. In future, we can revert this reflection based logic in the CodeStyle layer when there is no longer a need to clamp.